public/app/plugins/datasource/prometheus/querybuilder/shared/operationUtils.ts

```diff
@@ -144,3 +148,169 @@ export function getRangeVectorParamDef(withRateInterval = false): QueryBuilderOp
 
   return param;
 }
+
+/**
+ * This function is shared between Prometheus and Loki variants
+ */
+export function createAggregationOperation<T extends QueryWithOperations>(
+  name: string,
+  overrides: Partial<QueryBuilderOperationDef> = {}
+): QueryBuilderOperationDef[] {
+  const operations: QueryBuilderOperationDef[] = [
+    {
+      id: name,
+      name: getPromAndLokiOperationDisplayName(name),
+      params: [
+        {
+          name: 'By label',
+          type: 'string',
+          restParam: true,
+          optional: true,
+        },
+      ],
+      defaultParams: [],
+      alternativesKey: 'plain aggregations',
+      category: PromVisualQueryOperationCategory.Aggregations,
+      renderer: functionRendererLeft,
+      paramChangedHandler: getOnLabelAdddedHandler(`__${name}_by`),
+      explainHandler: getAggregationExplainer(name, ''),
+      addOperationHandler: defaultAddOperationHandler,
+      ...overrides,
+    },
+    {
+      id: `__${name}_by`,
+      name: `${getPromAndLokiOperationDisplayName(name)} by`,
+      params: [
+        {
+          name: 'Label',
+          type: 'string',
+          restParam: true,
+          optional: true,
+          editor: LabelParamEditor,
+        },
+      ],
+      defaultParams: [''],
+      alternativesKey: 'aggregations by',
+      category: PromVisualQueryOperationCategory.Aggregations,
+      renderer: getAggregationByRenderer(name),
+      paramChangedHandler: getLastLabelRemovedHandler(name),
+      explainHandler: getAggregationExplainer(name, 'by'),
+      addOperationHandler: defaultAddOperationHandler,
+      hideFromList: true,
+      ...overrides,
+    },
+    {
+      id: `__${name}_without`,
+      name: `${getPromAndLokiOperationDisplayName(name)} without`,
+      params: [
+        {
+          name: 'Label',
+          type: 'string',
+          restParam: true,
+          optional: true,
+          editor: LabelParamEditor,
+        },
+      ],
+      defaultParams: [''],
+      alternativesKey: 'aggregations by',
+      category: PromVisualQueryOperationCategory.Aggregations,
+      renderer: getAggregationWithoutRenderer(name),
+      paramChangedHandler: getLastLabelRemovedHandler(name),
+      explainHandler: getAggregationExplainer(name, 'without'),
+      addOperationHandler: defaultAddOperationHandler,
+      hideFromList: true,
+      ...overrides,
+    },
+  ];
+
+  return operations;
+}
+
+export function createAggregationOperationWithParam(
```

Comment:
The main change is this function. Before this code was inside `createAggregationOperation` and just special cased for topK/bottomK aggregations.